src/core/ext/filters/client_channel/subchannel.cc

```diff
@@ -332,7 +334,7 @@ class Subchannel::ConnectedSubchannelStateWatcher {
         case GRPC_CHANNEL_TRANSIENT_FAILURE:
         case GRPC_CHANNEL_SHUTDOWN: {
           if (!c->disconnected_ && c->connected_subchannel_ != nullptr) {
-            if (grpc_trace_stream_refcount.enabled()) {
+            if (grpc_trace_subchannel.enabled()) {
```

Comment:
Messages like this one actually have nothing to do with ref-counting and may be valuable in non-debug builds.  I suggest having two tracers here:

- "subchannel_refcount", which is debug-only
- "subchannel", which is in all builds

I suspect that we will want to add more trace logging for the latter over time.

Reply:
+1.

Note that you can add a TODO item for adding more trace logging for "subchannel". You can add more logging when you are working on your main project.